osutil/disks/size.go

```diff
@@ -33,15 +33,15 @@ import (
 func Size(partDevice string) (uint64, error) {
```

Comment:
Can this just use `	return blockDeviceSize(partDevice)` ? It seems we just duplicate this code right now? 
This:
```go
diff --git a/osutil/disks/blockdev.go b/osutil/disks/blockdev.go
index 5474ad0d1b..b515e0bfae 100644
--- a/osutil/disks/blockdev.go
+++ b/osutil/disks/blockdev.go
@@ -31,7 +31,7 @@ import (
 func blockdevSizeCmd(cmd, devpath string) (uint64, error) {
        out, err := exec.Command("blockdev", cmd, devpath).CombinedOutput()
        if err != nil {
-               return 0, osutil.OutputErr(out, err)
+               return 0, fmt.Errorf("cannot run blockdev: %v", osutil.OutputErr(out, err))
        }
        nospace := strings.TrimSpace(string(out))
        sz, err := strconv.ParseUint(nospace, 10, 64)
diff --git a/osutil/disks/size.go b/osutil/disks/size.go
index f7558e8e4b..a7af71de31 100644
--- a/osutil/disks/size.go
+++ b/osutil/disks/size.go
@@ -19,29 +19,10 @@
 
 package disks
 
-import (
-       "fmt"
-       "os/exec"
-       "strconv"
-       "strings"
-
-       "github.com/snapcore/snapd/osutil"
-)
-
 // Size returns the size of the given block device, e.g. /dev/sda1 in
 // bytes as reported by the kernels BLKGETSIZE ioctl.
 func Size(partDevice string) (uint64, error) {
        // Use blockdev command instead of calling the ioctl directly since
        // on 32bit systems it's a pain to get a 64bit value from a ioctl.
-       raw, err := exec.Command("blockdev", "--getsize64", partDevice).CombinedOutput()
-       if err != nil {
-               return 0, fmt.Errorf("cannot get disk size: %v", osutil.OutputErr(raw, err))
-       }
-       output := strings.TrimSpace(string(raw))
-       size, err := strconv.ParseUint(output, 10, 64)
-       if err != nil {
-               return 0, fmt.Errorf("cannot parse disk size output: %v", err)
-       }
-
-       return size, nil
+       return blockDeviceSize(partDevice)
 }
diff --git a/osutil/disks/size_test.go b/osutil/disks/size_test.go
index 7aeeef37cc..1f1c25826d 100644
--- a/osutil/disks/size_test.go
+++ b/osutil/disks/size_test.go
@@ -48,7 +48,7 @@ func (ts *sizeSuite) TestSizeErrFailure(c *C) {
        defer mockBlockdev.Restore()
 
        _, err := disks.Size("/dev/some-device")
-       c.Check(err, ErrorMatches, "cannot get disk size: some-error-message")
+       c.Check(err, ErrorMatches, "cannot run blockdev: some-error-message")
 }
 
 func (ts *sizeSuite) TestSizeErrSizeParsing(c *C) {
@@ -56,5 +56,5 @@ func (ts *sizeSuite) TestSizeErrSizeParsing(c *C) {
        defer mockBlockdev.Restore()
 
        _, err := disks.Size("/dev/some-device")
-       c.Check(err, ErrorMatches, `cannot parse disk size output: strconv.ParseUint: parsing "NaN": invalid syntax`)
+       c.Check(err, ErrorMatches, `cannot parse blockdev --getsize64 result size "NaN": strconv.ParseUint: parsing "NaN": invalid syntax`)
 }
```
seems to be work and removes some duplicated bits (but fine in a followup of course).

Reply:
I suspect it is dead code. I will try to remove it.

Reply:
It is dead. At least it builds locally. I have removed it. I will reorder the commits later when CI has managed to build it.